internal/lsp/server.go

```diff
@@ -492,16 +490,43 @@ func (l *LanguageServer) StartCommandWorker(ctx context.Context) {
 					break
 				}
 
-				responseParams := map[string]any{
-					"result": result,
-					"line":   line,
-				}
+				if l.clientIdentifier == clients.IdentifierVSCode {
+					responseParams := map[string]any{
+						"result": result,
+						"line":   line,
+					}
 
-				responseResult := map[string]any{}
+					responseResult := map[string]any{}
 
-				err = l.conn.Call(ctx, "regal/showEvalResult", responseParams, &responseResult)
-				if err != nil {
-					l.logError(fmt.Errorf("failed %s notify: %v", "regal/hello", err.Error()))
+					err = l.conn.Call(ctx, "regal/showEvalResult", responseParams, &responseResult)
+					if err != nil {
+						l.logError(fmt.Errorf("failed %s notify: %v", "regal/hello", err.Error()))
+					}
+				} else {
+					output := filepath.Join(workspacePath, "output.json")
+
+					var f *os.File
+
+					f, err = os.OpenFile(output, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0o755)
+
+					if err == nil {
+						var jsonVal []byte
+
+						value := result.Value
+						if result.IsUndefined {
+							// Display undefined as an empty object
+							// we could also go with "<undefined>" or similar
```

Comment:
I quite like the idea of calling it `undefined` rather than the empty object myself...

Reply:
I'm not too sure, as it makes it hard to differentiate from the actual string value of "undefined"... 
Using "<undefined>" as  `print` does is a little better. But for now I went with empty object as that's what `opa eval` prints by default, e.g.
```
opa eval input
{}
```